Writing Taylor & Francis proposal and updating Manning proposal (#417)
merely-useful/merely-useful.github.io: taylor-francis/taylor-francis-proposal-2020-07.md

# Building Research Software

-   Damien Irving
-   Kate Hertweck
-   Luke Johnston
-   Sara Mahallati
-   Joel Ostblom
-   Charlotte Wickham
-   Greg Wilson

## 1. Aims

Software is now as essential to research as telescopes, test tubes, and

> reference libraries, which means that researchers need need to know how to

Reviewer: ```suggestion
reference libraries, which means that researchers now need to know how to
```

Author: :+1:

build, check, use, and share programs.  However, most introductions to
programming focus on developing commercial applications, not on exploring
problems whose answers aren't yet known.  Our goal is show readers how to do
that, both on their own and as part of a team.

We believe every researcher should know how to write short programs that clean

> and analyze data in a reproducible way, and how to use version control to keep

Reviewer: Maybe just say "analyze data in a reproducible way" and leave out the reference to data cleaning, because that's not really a focus of this book (I'm assuming it will be in the novice book).

Author: :+1:

track of what they have done.  But just as some astronomers spend their careers
designing telescopes, some researchers focus on building the software that makes
research possible.  People who do this are called *research software engineers*,
and the aim of this book is to get you ready for this role, i.e., to help you go
from writing code for yourself to creating tools to help your entire field
advance.

## 2. Synopsis

This book introduces the core skills researchers need to develop robust software
that others can use, to share their work with others, and to be productive as
part of a research software team.  We assume readers have some basic programming
knowledge, and build on that to cover:

-   using the Unix shell to manage work and make it repeatable
-   effective use of version control
-   building reusable software tools
-   automating workflows
-   configuring software
-   testing and error handling
-   creating productive, inclusive teams

> -   publishing in a reproducible way

Reviewer: ```suggestion

```
Delete if proposal in #415 adopted.

Reviewer: We could add "documenting, packaging and releasing software" as the second last bullet point (above "creating productive, inclusive teams" which comes after packaging in the #415 proposal).

Author: :+1:

(e.g., Ray & Ray's *Unix and Linux: Visual QuickStart Guide*), are aimed at
commercial software developers (any number of books on testing, coding style,
and dev ops), or go into data science rather than telling readers how to build
software (e.g., Zhang's upcoming *A Tour of Data Science*).

Our book has broader and deeper coverage, and includes exercises with solutions.
Our focus on programming best practices (e.g. principles like writing modular,
reusable, testable code).  Others teach basic Python and R syntax and how to
complete isolated tasks with particular libraries, but today, people can pick up
that information from a Google search.  Our content is the bigger picture you
can't glean from online code snippets or Stack Overflow answers.

We go through the process of building a data science workflow/package from
scratch.  Most other texts simply present information (like a reference book):
we tackle a real data science problem from beginning to end, weaving in relevant
information as we go.

## 5. Format and Timeline

Our manuscript is approximately 470 pages long, with 40 diagrams and other
graphics and 150 code listing.

## 6. Other Information

> -   Dr. Damien B. Irving: climate scientist working for CSIRO in Tasmania.

Reviewer: ```suggestion
-   Dr. Damien B. Irving: post-doctoral researcher in climate science at the University of New South Wales.
```

Reviewer: (I'm still based in Tasmania as a visiting scientist at CSIRO, but my employer is now UNSW.)

Author: :+1:

    https://damienirving.github.io/

-   Dr. Kate L. Hertweck: bioinformatics training manager at the Fred Hutchinson
    Cancer Research Center. http://katehertweck.com/

-   Dr. Luke Johnston: post-doctoral researcher in bioinformatics at Aarhus
    University. http://lukewjohnston.com/

-   Sara Mahallati: graduate student in biomedical engineering at the University
    of Toronto. https://scholar.google.ca/citations?user=8zCoa-UAAAAJ

-   Joel Ostblom: graduate student in biomedical engineering at the University
    of Toronto. https://joelostblom.com/

-   Dr. Charlotte Wickham: a data scientist and instructor at Oregon State
    University. https://www.cwick.co.nz/

-   Dr. Gregory V. Wilson: co-founder of Software Carpentry, now part of the
    education group at RStudio. http://third-bit.com/

> ## Table of Contents

Reviewer: The table of contents might change as a result of discussions in #415

Author: Yup.
